doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -6,5 +6,131 @@ nav: /:path/Programming-guide/Key-platform-features/Devices-provisioning-and-reg
 sort_idx: 10
 ---
 
-Devices provisioning and registration
+* TOC
+{:toc}
 
+{% assign root_url = page.url | split: '/'%}
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
+
+Kaa establishes a mechanism to uniquely identify each endpoint and the associated tokens and credentials issued to that endpoint.
+Before first connection to Kaa you should generate a key pair (private and public key) and send request for registration of public key in Kaa. 
+For this use [Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#TODO).
+Registration of public keys occurs only once. In such an approach, all message will be encrypted with a recipient's public key. 
+The message cannot be decrypted by anyone who does not possess the matching private key, who is thus presumed to be the owner of that key and the endpoint associated with the public key. 
+Also you can disable the access to server for endpoint by terminate all active sessions with this endpoint. All subsequent requests will be forbidden.
+For this use [Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#TODO).
+
+## Credentials service
+
+Credential is an object that is verified when presented to the verifier in an authentication transaction. 
+It establish the identity of endpoint. 
+
+Kaa provides two credentials service
+
+1. Trustful credentials service -  allows any endpoint to register and connect to Kaa cluster. (like the previous version of Kaa)
+2. Internal credentials service -  allows connecting with Kaa for specified list of endpoints whose credentials was previously provisioned with Kaa REST API.
+
+When Tenant Admin create application he can specify credential service for this application. By default used "Trustful".
+
+## Creating custom credentials service
+
+1. First of all you should create class which implements all method of CredentialsService interface
```

Comment:
which implements CredentialsService(link to github file in master branch) interface